# STAT 5010 Final Project
# NBA Analytics using R

## Done by:
- **Pranav Unnikrishnan** [111370740]
- **Rishekesan Senthil Kumar Vanathi** [111350847]


### Hypothesis Test 1


 Do Point Guards (PG) score fewer points than Shooting Guards (SG)?
$H_0$: $\mu_{\text{PG}} = \mu_{\text{SG}}$


$H_1$: $\mu_{\text{PG}} < \mu_{\text{SG}}$

In [64]:
# Load the dataset
data <- read.csv("C:\\grad_sem2\\stats2\\Project\\NBA-Analytics-Using-R\\Data\\Processed\\nba_normal_scorers_filtered.csv")

# Basic dataset overview
library(dplyr)
library(ggplot2)



In [66]:


pg_pts <- subset(data, Pos == "PG")$PTS
sg_pts <- subset(data, Pos == "SG")$PTS

t.test(pg_pts, sg_pts, alternative = "less", var.equal = FALSE)



	Welch Two Sample t-test

data:  pg_pts and sg_pts
t = 0.72701, df = 698.62, p-value = 0.7663
alternative hypothesis: true difference in means is less than 0
95 percent confidence interval:
      -Inf 0.9839645
sample estimates:
mean of x mean of y 
 7.818788  7.517467 


####  **Interpretation**

**Test Type:** Welch Two Sample *t*-test (unequal variances)

- Test Statistic: $t = 0.727$  
- Degrees of Freedom: $df = 698.62$  
- p-value: $0.7663$  
- 95% Confidence Interval: $(-\infty,\ 0.984)$  
- Mean Points (PG): $7.82$  
- Mean Points (SG): $7.52$

Since the p-value ($0.7663$) is **much greater** than $0.05$, we **fail to reject the null hypothesis**.

There is **no statistically significant evidence** that Point Guards score fewer points than Shooting Guards.


### Hypothesis testing 2

 ##### Is there a difference in average salary between players selected as All-Stars and non-All-Stars?

$H_0$: $\mu_{\text{AllStar}} = \mu_{\text{NonAllStar}}$



$H_1$: $\mu_{\text{AllStar}} \ne \mu_{\text{NonAllStar}}$

In [68]:
table_pos_allstar <- table(data$Pos, data$AllStarSelection)
chisq.test(table_pos_allstar)



	Pearson's Chi-squared test

data:  table_pos_allstar
X-squared = 9.1613, df = 4, p-value = 0.05719


#### **Interpretation**

**Test Type:** Pearson's Chi-Square Test of Independence

- Test Statistic: $\chi^2 = 9.161$  
- Degrees of Freedom: $df = 4$  
- p-value: $0.0572$

Since the p-value ($0.0572$) is **slightly above** the common significance level of $0.05$, we **fail to reject the null hypothesis** at the 5% level.

However, the result is **close to the threshold**, suggesting a **possible weak association** between player position and All-Star selection.

A more lenient significance level (e.g., $\alpha = 0.1$) or a larger dataset may indicate a statistically meaningful relationship.


### Hypothesis Test 3: 


#### Do assists significantly predict player salary after accounting for minutes played and position?



$H_0$: The coefficient for assists ($\beta_{\text{AST}}$) = 0

$H_1$: $\beta_{\text{AST}} \ne 0$ (Assists contribute significantly to salary

In [71]:


# Clean up: drop NA, convert position to factor
df <- subset(data, !is.na(Salary) & !is.na(AST) & !is.na(MP))
df$Pos <- factor(df$Pos)

# Multiple Linear Regression
model <- lm(Salary ~ AST + MP + Pos, data = df)
summary(model)  # Check coefficients

# ANOVA to test overall model significance
anova(model)



Call:
lm(formula = Salary ~ AST + MP + Pos, data = df)

Residuals:
      Min        1Q    Median        3Q       Max 
-26338860  -4008820   -780782   2480247  36231891 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  -429742     531243  -0.809  0.41867    
AST          2116213     181441  11.663  < 2e-16 ***
MP            354289      28818  12.294  < 2e-16 ***
PosPF       -1534578     548732  -2.797  0.00522 ** 
PosPG       -4471538     659861  -6.776 1.70e-11 ***
PosSF       -3038644     554321  -5.482 4.86e-08 ***
PosSG       -4195628     529371  -7.926 4.12e-15 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 7003000 on 1662 degrees of freedom
Multiple R-squared:  0.4356,	Adjusted R-squared:  0.4335 
F-statistic: 213.8 on 6 and 1662 DF,  p-value: < 2.2e-16


,Df,Sum Sq,Mean Sq,F value,Pr(>F)
,<int>,<dbl>,<dbl>,<dbl>,<dbl>
AST,1,4.675696e+16,4.675696e+16,953.34438,7.728929e-166
MP,1,1.220267e+16,1.220267e+16,248.80468,2.433024e-52
Pos,4,3.943473e+15,9.858683e+14,20.10122,3.502720e-16
Residuals,1662,8.151311e+16,4.904519e+13,NA,NA


#### **Interpretation**

The multiple linear regression shows that:

- **Assists per game** ($\text{AST}$) have a **strong, statistically significant** positive effect on player salary, even after accounting for minutes played and position.

- The coefficient for assists is approximately **\$2,116,213**, with a p-value $< 2 \times 10^{-16}$, indicating that:

$$
\text{Salary increases by approximately \$2.1M for each additional assist per game}
$$

- **Minutes played per game** ($\text{MP}$) is also a significant predictor, with an estimated increase of **\$354,289** per additional minute.  
  This supports the idea that more playing time leads to greater compensation.

- **Position** ($\text{Pos}$) is a significant categorical variable. Compared to the baseline group (likely Centers), the coefficients for PG, SG, and SF are significantly negative, suggesting these roles are paid less on average:

$$
\text{PG, SG, and SF} \longrightarrow \text{Lower salary compared to baseline (e.g., C)}
$$

- The model’s **Adjusted** $R^2 = 0.4335$, meaning about **43.4%** of the variation in salary can be explained by assists, minutes played, and position.


#### **Conclusion**

Assists and minutes per game are **strong predictors** of NBA player salary.

Even after controlling for position, each assist and each additional minute played contribute significantly to higher earnings.

Additionally, position plays an important role — **Point Guards, Shooting Guards, and Small Forwards** are compensated less on average compared to players in the baseline position group (likely Centers).

This model captures both performance metrics and role-based pay dynamics, and explains a substantial **43.4%** of the variation in player salaries.


### Hypothesis Test 4

#### Do All-Star Players Earn More Than Non-All-Stars?

$H_0$: Mean salary of All-Stars = Mean salary of non-All-Stars

$H_1$: Mean salary of All-Stars ≠ Mean salary of non-All-Stars



In [74]:
as_salary <- subset(data, AllStarSelection == 1)$Salary
nas_salary <- subset(data, AllStarSelection == 0)$Salary

t.test(as_salary, nas_salary, alternative = "two.sided", var.equal = FALSE)



	Welch Two Sample t-test

data:  as_salary and nas_salary
t = 10.304, df = 70.067, p-value = 1.122e-15
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 14166934 20967465
sample estimates:
mean of x mean of y 
 23602322   6035122 


#### **Interpretation**

The p-value is far below any conventional significance threshold, so we **strongly reject the null hypothesis**.

There is compelling statistical evidence that **NBA All-Star players earn significantly more** than their non-All-Star counterparts.

On average, the salary difference between the two groups is approximately **\$17.5 million**, with the true gap likely falling within the 95% confidence interval:

$$
17.5\text{M} \in [14.1\text{M},\ 20.9\text{M}]
$$


#### **Conclusion**

Being selected as an **NBA All-Star** is associated with **substantially higher salaries**.

This reflects more than just on-court performance — it also captures:

- **Marketability**  
- **Star power**  
- **Franchise value**  
- **Elite status** within the league


### Hypothesis Test 5

#### Is Scoring Variability Equal Across NBA Teams?

$H_0$: All teams have the same variance in player points per game ($\sigma^2_{\text{Team A}} = \sigma^2_{\text{Team B}} = \dots$)

$H_1$: At least one team has a different variance in scoring than the others

In [75]:
# Load required library
library(car)

# Remove NA and filter valid team/player PTS values
df_levene <- subset(data, !is.na(PTS) & !is.na(Team))

# Levene's Test of Homogeneity of Variance
leveneTest(PTS ~ Team, data = df_levene)


Warning message in leveneTest.default(y = y, group = group, ...):
"group coerced to factor."


,Df,F value,Pr(>F)
,<int>,<dbl>,<dbl>
group,32,2.021204,0.0006449633
,1837,NA,NA


#### **Interpretation**

The p-value is:

$$
p = 0.000645
$$

This value is well below the conventional significance level of:

$$
\alpha = 0.05
$$

Therefore, we reject the null hypothesis:

$$
H_0: \sigma^2_{\text{Team}_1} = \sigma^2_{\text{Team}_2} = \dots = \sigma^2_{\text{Team}_{33}}
$$

There is strong statistical evidence that **not all NBA teams have equal scoring variability** among their players.



#### **Conclusion**

Some NBA teams demonstrate significantly more **scoring imbalance or volatility** than others.

This difference may be due to:

- Dependence on one or two superstar scorers  
- Balanced vs. top-heavy roster structure  
- Variation in offensive distribution strategies or coaching philosophy
